src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -299,7 +299,7 @@ public HttpConnection(Uri url, ITransportFactory transportFactory, ILoggerFactor
                     using (var response = await httpClient.SendAsync(request))
```

Comment:
You can make a small optimization here to avoid the internal buffer but passing HttpCompletionOption.ResponseHeadersRead to SendAsync

Reply:
Will leave for a follow up PR. I wanna get this in.